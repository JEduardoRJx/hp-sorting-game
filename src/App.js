import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import './App.scss'

export class App extends Component {
  constructor() {
    super();
  }

  render() {
    return (
      <main>
        Harry Potter and the Sorting Game
      </main>
    )
  }
}