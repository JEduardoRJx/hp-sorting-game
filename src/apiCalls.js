const baseUrl = 'https://www.potterapi.com/v1';

const apiKey = `?key=$2a$10$soGCoAQkdenPDKuyeTco3urLugvc4a/RMC2S3tzviQxAbVxmHAPN6`

const apiKey2 = `?key=$2a$10$AZk7aQ1VZuDe/U1v8A.9u.k/U7knzlj8B7gonTU6Qw/mtqV4rO6zq`

export const fetchCharacters = async () => {
  const url = `${baseUrl}/characters${apiKey2}`;
  const response = await fetch(url);
  if(!response.ok) {
    throw Error('There\'s an error with your wand, can\'t fetch characters');
  }
  const characters = await response.json();
  return characters;
}

export const fetchSortingHat = async () => {
  const url = `${baseUrl}/sortingHat`;
  const response = await fetch(url);
  if(!response.ok) {
    throw Error('The sorting hat must me asleep, can\'t fetch House');
  }
  const house = await response.json();
  return house;
}