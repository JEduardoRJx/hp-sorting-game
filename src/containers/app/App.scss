$hw-dark: #4c3638;
$hw-light: #D3A481;
$sl-dark: #276F45;
$sl-light: #A8B0BF;
$gryf-dark: #702737;
$gryf-light: #DAA520;
$rav-dark: #274F70;
$rav-light: #C98C4E;
$huff-dark: #141414;
$huff-light: #D1B738;

@font-face {
  font-family: 'harry_pregular';
  src: url('../../fonts/harryp__-webfont.woff') format('woff'),
       url('../../fonts/harryp__-webfont.woff2') format('woff2');
  font-weight: normal;
  font-style: normal;
}

* {
  margin: 0px;
  padding: 0px;
  transition: ease all .3s;
}

main {
  background-color: $hw-dark;
  height: 100vh;
  overflow: scroll;
}

.game-title {
  font-family: 'harry_pregular';
  font-size: 4.5em;
  height: 10vh;
  color: $hw-light;
  text-align: center;
  border-bottom: 3px solid $hw-light;
  
}