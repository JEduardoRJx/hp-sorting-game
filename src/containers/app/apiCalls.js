const baseUrl = 'https://www.potterapi.com/v1';
const sortingHat = '/sortingHat';
const houses = '/houses'
const spells = '/spells'

const apiKey = `?key=$2a$10$soGCoAQkdenPDKuyeTco3urLugvc4a/RMC2S3tzviQxAbVxmHAPN6`

export const fetchCharacters = async () => {
  const url = `${baseUrl}/characters${apiKey}`;
  const response = await fetch(url);
  if(!response.ok) {
    throw Error('There\'s an error with your wand, can\'t fetch characters');
  }
  const characters = await response.json();
  return characters;
}