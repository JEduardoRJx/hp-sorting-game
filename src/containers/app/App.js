import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import './App.scss';
import { connect } from 'react-redux';
import { fetchCharacters } from './apiCalls'
import Modal from '../../components/modal/Modal'

export class App extends Component {
  constructor() {
    super();
  }

  // async componentDidMount() {
  //   try {
  //     const characters = await fetchCharacters();
  //     //action that sets characters to state
  //     console.log(characters)
  //     //action that filters characters set to a house
  //     const charactersInHouses = characters.filter(char => 'house' in char)
  //     //set charactersInHouses to state
  //     console.log(charactersInHouses)
  //   } catch({ message }) {
  //     console.log(message);
  //   }
    
  // }

  closeModal = () => {
    const modal = document.querySelector('.modal');
    const modalTitle = document.querySelector('.modal-title');
    const modalImg = document.querySelector('.modal-img');
    const modalGif = document.querySelector('.modal-gif');
    modalTitle.style.display = 'none';
    modalImg.style.display = 'none';
    modalGif.style.display = 'block'
    setTimeout(function(){ 
      modal.style.display = 'none';
    }, 3000);
  }

  render() {
    return (
      <main>
        <h1 className='game-title'>Harry Potter and the Sorting Game</h1>
        <Modal closeModal={this.closeModal}/>
      </main>
    )
  }
}

export const mapStateToProps = ({}) => ({

})

export const mapDispatchToProps = dispatch => ({

});

export default connect(mapStateToProps, mapDispatchToProps)(App);