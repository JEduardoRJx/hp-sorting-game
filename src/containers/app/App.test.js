import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App'
import { fetchCharacters } from '../../apiCalls'
jest.mock('../../apiCalls');

describe('AppContainer', () => {

  describe('App component', () => {
    let wrapper;
    const mockSetCharacters = jest.fn();
    const mockLoading = jest.fn();
    const mockCharacters = [{
      bloodStatus: "pure-blood",
      deathEater: false,
      dumbledoresArmy: false,
      house: "Hufflepuff",
      ministryOfMagic: false,
      name: "Cedric Diggory",
      orderOfThePhoenix: false,
      role: "Student",
      school: "Hogwarts School of Witchcraft, and Wizardry",
      species: "human",
      wand: 'Ash, 12 1/4", unicorn hair',
      __v: 0,
      _id: "5a1096253dc2080021cd875f"
    }]
    const mockIsLoading = true;
    const mockUser = {
      name: 'name',
      wizardkind: 'wizard',
      wand: 'wand',
      house: 'slytherin'
    }
    fetchCharacters.mockImplementation(() => Promise.resolve(mockCharacters))

    beforeEach(() => {
      wrapper = shallow (<App 
      setCharacters={mockSetCharacters}
      loading={mockLoading}
      allCharacters={mockCharacters}
      isLoading={true}
      user={mockUser}
      />)
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should setCharacters when fetchCharacters is called', () => {
      expect(fetchCharacters).toHaveBeenCalled();
      expect(mockSetCharacters).toHaveBeenCalledWith(mockCharacters);
      expect(mockLoading).toHaveBeenCalledWith(false);
    });

  });

  describe('mapStateToProps', () => {

  });

  describe('mapDispatchToProps', () => {
    
  })
})