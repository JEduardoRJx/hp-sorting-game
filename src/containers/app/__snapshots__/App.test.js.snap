// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppContainer App component should match snapshot 1`] = `
<main
  className="slytherin"
>
  <h1
    className="game-title slytherin"
  >
    Harry Potter and the Sorting Game
  </h1>
  <Redirect
    to="/"
  />
  <Modal
    closeModal={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/game"
    render={[Function]}
  />
  <Route
    path="/discoveryourhouse"
    render={[Function]}
  />
</main>
`;
